### PR TITLE
feat: manage APIs for user grants

### PR DESCRIPTION
**Pull request by kpears201**

### CODE REVIEW

File: src/modules/user_grants.json [hunk 1]
```diff
@@ -0,0 +1,534 @@
...
+                        "value": [
+                            {
+                                "app": {
+                                    "id": "certapp",
+                                    "title": "Firebolt Certification"
+                                },
+                                "state": "active",
+                                "capability": "xrn:firebolt:capability:data:app-usage",
+                                "role": "use",
+                                "lifespan": "seconds",
```
Comment (satlead) [inline src/modules/user_grants.json:59]: Does lifespan here mean that it is time bound?

File: src/modules/user_grants.json [hunk 2]
```diff
@@ -0,0 +1,534 @@
...
+                                "lifespan": "powerActive"
+                            }
+                        ]
+                    }
+                }
+            ]
+        },
+        {
+            "name": "capability",
+            "summary": "Get all active and denied user grants for the given capability",
```
Comment (satlead) [inline src/modules/user_grants.json:119]: Wouldn't this be per app?

Comment (satlead) [reply src/modules/user_grants.json:119]: A good example
>"Give me all the apps I have granted access to the camera"

**Review by satlead — DISMISSED** [review]

File: src/modules/user_grants.json [hunk 3]
```diff
@@ -0,0 +1,534 @@
...
+                "additionalProperties": false,
+                "required": [
+                    "id"
+                ]
+            },
+            "GrantState": {
+                "description": "The state the grant is in",
+                "type": "string",
+                "enum": [
+                    "active",
```
Comment (satlead) [inline src/modules/user_grants.json:517]: active seems more implicit in terms of saying its granted + alive . But then we are trying to push lifespan into a state wouldn't that mean we need to carry state like expired?

File: src/modules/user_grants.json [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+                    "denied"
+                ]
+            },
+            "GrantModificationOptions": {
+                "description": "Options when modifying any grant",
+                "type": "object",
+                "properties": {
+                    "appId": {
```
Comment (satlead) [inline src/modules/user_grants.json:525]: Seem to have only appId are we going to have more props in future feels appId seems more appropriate given grant/deny and request definitely need appId

Comment (kpears201) [reply src/modules/user_grants.json:525]: > appId seems more appropriate given grant/deny and request definitely need appId

grant and deny do not need appId if it is a device-scoped grant.

Using an options object so it is extensible in the future, but currently only appId is optional.

File: src/modules/user_grants.json (hunk 3 above)
Comment (kpears201) [reply src/modules/user_grants.json]: Changed active to "granted"

**Review by satlead — APPROVED** [review]
LGTM

**Pull request merged**